analyses/rna-seq-composition/rna-seq-composition.Rmd

```diff
@@ -0,0 +1,579 @@
+
+# Check whether all samples are present in both
+table(star_results_w_biospec$Kids.First.Biospecimen.ID %in% mend_results_w_biospec$Kids.First.Biospecimen.ID)
+table(mend_results_w_biospec$Kids.First.Biospecimen.ID %in% star_results_w_biospec$Kids.First.Biospecimen.ID)
+
+
+# Combine results from different outputs
+comb_results <- full_join(star_results_w_biospec, 
+                          mend_results_w_biospec, 
+                          by="Kids.First.Biospecimen.ID") %>%
+  select(-star_results_file, -MEND_results_file, -qc, -Sample) %>%
+  rename(sample_id = Kids.First.Biospecimen.ID) 
+
+```
+
+
+# Groom data 
+
+```{r}
+
+read_counts <- comb_results %>%
+  mutate(Mapped = Multimapped + UM_reads) %>%
+  select(-UM_reads) 
+
+
+samples_in_depth_order <- read_counts$sample_id[order(desc(read_counts$Total_reads))]
+
+
+```
+
+
+# Filter for library prep method
+```{r}
+
+metadata_raw <- read_tsv(file.path(data_dir, "pbta-histologies.tsv"), 
+                     col_types = cols(
+                       .default = col_character())) 
+
+metadata <- metadata_raw %>%
+  filter(experimental_strategy == "RNA-Seq") %>%
+  select(Kids_First_Biospecimen_ID, RNA_library, composition, sample_type)
+
+# How many samples of each type are present?
+metadata %>% tabyl(RNA_library, composition, sample_type)
+
+stranded_samples <- metadata %>% 
+  filter(RNA_library == "stranded") %>%
+  pull(Kids_First_Biospecimen_ID)
+
+selected_read_counts <- read_counts %>%
+  filter(sample_id %in% stranded_samples)
+```
+
+This analysis is performed only on ribo-deplete data (stranded). Data from cell lines and solid tissues are both included. 
+
+# Calculate percentages for reference range calcs
+```{r}
+
+read_counts_with_ref_range_fractions <- selected_read_counts %>% 
+  arrange(desc(Total_reads)) %>% 
+  mutate(pct_unmapped_of_total = (Total_reads - Mapped)/Total_reads,
+         pct_multi_mapped_of_mapped = Multimapped/Mapped, # Mapped = UM + MM
+         pct_dupe_of_mapped = (Mapped - MND) / Mapped,
+         pct_non_exonic_of_non_dupe = (MND - MEND)/MND)
+
+read_fracs_with_ref_range_fractions_long <- read_counts_with_ref_range_fractions %>%
+  select(c(sample_id, starts_with("pct_"))) %>% 
+  gather(key="ref_range_name", value = "value", -sample_id) 
+
+```
+
+# Calculate reference ranges
+```{r}
+
+ref_range_codes <- as_tibble(
+  list(ref_range_abbrev = c("NM", "MM", "D", "NE"), 
+       ref_range_description = 
+         c("Not mapped", "Multi-mapped", "Duplicate reads", "Non exonic reads"), 
+       ref_range_name = 
+         c("pct_unmapped_of_total", "pct_multi_mapped_of_mapped", 
+           "pct_dupe_of_mapped", "pct_non_exonic_of_non_dupe"
+         )))
+
+ref_range_stats <- function(mean1, sd1) {
+  c(mean_minus_one_sd = mean1-sd1,
+    mean_plus_one_sd = mean1+sd1,
+    mean_minus_two_sd = mean1-(2*sd1),
+    mean_plus_two_sd = mean1+(2*sd1))}
+
+ref_range_mean_sd <- read_counts_with_ref_range_fractions %>%
```

Comment:
I'm adding some comments. Do you think this would also be a useful place for a couple head statements or is that overkill?

I like the approach of using more straightforward data structures (rather than nested ones), but I don't understand how to implement your suggestion here.

Reply:
I think it would be more of a refactor, but the steps would be these: 
1. group by ref_range_name
2. summarise mean and sd
3. mutate to add mean_plus and mean_minus columns

Actually, I think that is it in this case. Maybe it was another case where I was thinking about with join? Or maybe I got myself confused here.